components/select/select-option.tsx

```diff
@@ -52,20 +54,40 @@ const SelectOption: React.FC<React.PropsWithChildren<SelectOptionProps>> = ({
   }, [identValue, value])
 
   const bgColor = useMemo(() => {
-    if (isDisabled) return theme.palette.accents_1
-    return selected ? theme.palette.accents_2 : theme.palette.background
+    if (isDisabled) return theme.palette.cNeutral3
+    return selected && variant !== 'text' ? theme.palette.cTheme5 : theme.palette.cNeutral8
   }, [selected, isDisabled, theme.palette])
 
+  const rgb = useRgb(theme.palette.cTheme5)
   const hoverBgColor = useMemo(() => {
     if (isDisabled || isLabel || selected) return bgColor
-    return theme.palette.accents_1
+    return `rgba(${rgb.r}, ${rgb.g}, ${rgb.b}, 0.04)`
   }, [selected, isDisabled, theme.palette, isLabel, bgColor])
 
   const color = useMemo(() => {
-    if (isDisabled) return theme.palette.accents_4
-    return selected ? theme.palette.foreground : theme.palette.accents_5
+    if (isDisabled) return theme.palette.cNeutral4
+    return selected
+      ? variant === 'text'
+        ? theme.palette.cTheme5
+        : theme.palette.cNeutral8
+      : theme.palette.cNeutral6
```

Comment:
Better use `if...else` or something more clear here.

Reply:
the code has been moved to style.ts